community/kernel/src/main/java/org/neo4j/unsafe/batchinsert/internal/BatchInserterImpl.java

```diff
@@ -1105,46 +1112,63 @@ void forceFlushChanges()
 
     private class BatchSchemaActions implements InternalSchemaActions
     {
+        private int[] getOrCreatePropertyKeyIds( Iterable<String> properties )
+        {
+            ArrayList<Integer> propertyKeyIds = new ArrayList<>();
+            properties.forEach( index -> propertyKeyIds.add( getOrCreatePropertyKeyId( index ) ) );
+            return propertyKeyIds.stream().mapToInt( i -> i ).toArray();
+
+        }
+
+        private int[] getOrCreatePropertyKeyIds( String[] propertyKeys )
+        {
+            return getOrCreatePropertyKeyIds( Arrays.asList( propertyKeys ) );
```

Comment:
Even if it perhaps doesn't matter since not very many rules are created, this is production code, after all and there's some hefty garbage making going on here: String[] --> List<String> --> List<String> (again, also creating a lambda instance) --> Stream --> String[]. We should be able to do better, it's becomes a bad habit to litter too much.